swarm/src/handler/one_shot.rs

```diff
@@ -213,7 +213,9 @@ where
             }
             ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
                 if self.pending_error.is_none() {
+                    log::debug!("DialUpgradeError: {error:?}");
```

Comment:
```suggestion
                    log::debug!("DialUpgradeError: {error}");
```

Errors can be display formatted but at the moment, that does not necessarily print the entire chain of errors.

@mxinden Should we depend on `anyhow` internally here to print all errors?

Reply:
There is a function `print_error_chain` in `swarm/src/lib.rs`. I could use that one. Is this an option?

Reply:
Happy to use that one yeah as we are within the same crate!